src/action/films.js

```diff
@@ -1,26 +1,26 @@
-import { SET_FILMS, SET_SORT, SET_GENRE } from "./action-types";
+import { DEFAULT_SORT_FIELD } from "../Constants";
+import { SET_FILMS } from "./action-types";
 
 export const setFilms = (list) => ({
   type: SET_FILMS,
   payload: list,
 });
 
-export const fetchFilms = (count) => {
-  return (dispatch, getState) => {
-    const { sort, genre } = getState().films;
-    let url = `http://localhost:4000/movies?limit=${count}&sortBy=${sort}&sortOrder=desc`;
+export const fetchFilms = (count, location, searchQuery) => {
+  return (dispatch) => {
+    const query = new URLSearchParams(location.search);
+    const sort = query?.get("sortBy") || DEFAULT_SORT_FIELD;
+    const sortOrder = sort === "title" ? "asc" : "desc";
+    const genre = query?.get("genre");
+    let url = `http://localhost:4000/movies?limit=${count}&sortBy=${sort}&sortOrder=${sortOrder}`;

```

Comment:
Not necessary to do❗
`nit`: You may split this part into separate constants and combine them into `url` before fetching. 
```js
const DEFAULT_URL = 'http://localhost:4000'
```